stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/controller/SendMessageOfflineTest.kt

```diff
@@ -0,0 +1,103 @@
+package io.getstream.chat.android.livedata.controller
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argThat
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.api.models.WatchChannelRequest
+import io.getstream.chat.android.client.channel.ChannelClient
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.livedata.repository.RepositoryFacade
+import io.getstream.chat.android.livedata.utils.TestDataHelper
+import io.getstream.chat.android.test.InstantTaskExecutorExtension
+import io.getstream.chat.android.test.TestCall
+import io.getstream.chat.android.test.TestCoroutineExtension
+import io.getstream.chat.android.test.getOrAwaitValue
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runBlockingTest
+import org.amshove.kluent.`should be equal to`
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.RegisterExtension
+
+@ExperimentalCoroutinesApi
+@ExtendWith(InstantTaskExecutorExtension::class)
+internal class SendMessageOfflineTest {
+
+    @JvmField
+    @RegisterExtension
+    val testCoroutines = TestCoroutineExtension()
+
+    private val data = TestDataHelper()
```

Comment:
Could we use "Mother methods" instead of this class?
The mother method create random values everytimes